src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java

```diff
@@ -385,6 +386,8 @@ static String getResource(String key) {
             {"R_UnknownFeatureAck", "Unknown feature acknowledge is received."},
             {"R_SetAuthenticationWhenIntegratedSecurityTrue",
                     "Cannot set \"Authentication\" with \"IntegratedSecurity\" set to \"true\"."},
+            {"R_NoUserPasswordForDomain",
+                    "Both \"User\" (or \"UserName\") and \"Password\" connection string keywords must be specified for \"domain\"."},
```

Comment:
I have a few concerns regarding this error message.
- The driver throws this error message even if both username and password are present, but domain name is missing. (`authentication=NTLM;username=user;password=password;`). I am suggesting to modify the error message to something along `User, Password, domain must be specified for NTLM authentication.`
- `connection string keywords` does not sound correct, because these properties can be specified through SQLServerDataSource too. Perhaps `Datasource/Connection String keywords` is a better option.